<launch>
<arg name="scan_topic" default="scan"/>
    <node pkg="amcl" type="amcl" name="amcl" output="screen">
        <remap from="scan" to="scan_filtered"/>

        <!-- Overall filter parameters -->
        <param name="min_particles" value="500"/>
        <param name="max_particles" value="2000"/>
        <param name="kld_err" value="0.05"/>
        <param name="kld_z" value="0.99"/>
        <param name="update_min_d" value="0.25"/> <!-- Translational movement required before performing a filter update. -->
        <param name="update_min_a" value="0.2"/> <!-- Rotational movement required before performing a filter update. 0.1 represents 5.7 degrees  -->
        <param name="resample_interval" value="1"/> <!-- Number of filter updates required before resampling. -->
        <param name="transform_tolerance" value="0.1"/> <!-- Default 0.1; time with which to post-date the transform that is published, to indicate that this transform is valid into the future. -->
        <param name="recovery_alpha_slow" value="0.001"/> <!-- Exponential decay rate for the slow average weight filter, used in deciding when to recover by adding random poses. -->
        <param name="recovery_alpha_fast" value="0.1"/> <!-- Exponential decay rate for the fast average weight filter, used in deciding when to recover by adding random poses. -->
        <param name="gui_publish_rate" value="10.0"/> <!-- Maximum rate (Hz) at which scans and paths are published for visualization, -1.0 to disable. -->
        <param name="save_pose_rate"   value="0.3"/>
        <param name="use_map_topic"    value="false"/>
        <param name="first_map_only"   value="false"/>

        <!-- set particles init pose to robot pose -->
        <param name="initial_pose_x" value="0.0"/> <!-- -0.5 -->
        <param name="initial_pose_y" value="0.0"/>
        <param name="initial_pose_a" value="0.0"/>
        <param name="initial_cov_xx" value="0.0"/>
        <param name="initial_cov_yy" value="0.0"/>
        <param name="initial_cov_aa" value="0.0"/> <!-- pi/12 * pi/12 0.033-->

        <!-- Laser model parameters -->
	<remap from="scan" to="$(arg scan_topic)"/>
	    <param name="laser_min_range" value="-1" />
	    <param name="laser_max_range" value="12" />
        <param name="laser_max_beams" value="60"/>	
        <param name="laser_z_hit" value="0.5"/>
        <param name="laser_z_short" value="0.05"/>
        <param name="laser_z_max" value="0.05"/>
        <param name="laser_z_rand" value="0.5"/>
        <param name="laser_sigma_hit" value="0.2"/>
        <param name="laser_lambda_short" value="0.1"/>
        <param name="laser_likelihood_max_dist" value="2.0"/>
        <param name="laser_model_type" value="likelihood_field"/>

        <!-- Odometry model parameters -->
        <param name="odom_model_type" value="omni-corrected"/>
        <param name="odom_alpha1" value="0.2"/> <!-- Specifies the expected noise in odometry's rotation estimate from the rotational component of the robot's motion. -->
        <param name="odom_alpha2" value="0.2"/> <!-- Specifies the expected noise in odometry's rotation estimate from translational component of the robot's motion. -->
        <param name="odom_alpha3" value="0.2"/> <!-- Specifies the expected noise in odometry's translation estimate from the translational component of the robot's motion. -->
        <param name="odom_alpha4" value="0.2"/> <!-- Specifies the expected noise in odometry's translation estimate from the rotational component of the robot's motion. -->
        <param name="odom_alpha5" value="0.1"/> <!--only used if model is "omni"-->
        <param name="odom_frame_id" value="odom_combined"/>
        <param name="base_frame_id" value="base_footprint"/> <!-- Change this if you want to change your base frame id. -->
        <param name="global_frame_id" value="map"/>
	    <param name="tf_broadcast"    value="true"/>
    </node>
</launch>